msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AuthorizationCodeRequest.java

```diff
@@ -17,19 +17,19 @@ class AuthorizationCodeRequest extends MsalRequest {
     }
 
     private static AbstractMsalAuthorizationGrant createMsalGrant(AuthorizationCodeParameters parameters) {
+        Map<String, List<String>> params = new LinkedHashMap<>();
+
+        params.put(GrantConstants.GRANT_TYPE_PARAMETER, Collections.singletonList(GrantConstants.AUTHORIZATION_CODE));
+        params.put("code", Collections.singletonList(parameters.authorizationCode()));
+
+        if (parameters.redirectUri() != null) {
+            params.put("redirect_uri", Collections.singletonList(parameters.redirectUri().toString()));
+        }
 
-        AuthorizationGrant authorizationGrant;
         if (parameters.codeVerifier() != null) {
-            authorizationGrant = new AuthorizationCodeGrant(
-                    new AuthorizationCode(parameters.authorizationCode()),
-                    parameters.redirectUri(),
-                    new CodeVerifier(parameters.codeVerifier()));
-
-        } else {
-            authorizationGrant = new AuthorizationCodeGrant(
-                    new AuthorizationCode(parameters.authorizationCode()), parameters.redirectUri());
+            params.put("code_verifier", Collections.singletonList(parameters.codeVerifier()));
```

Comment:
I am not sure how much of PKCE was handled by nimbus, so please make sure we have enough tests that validate that auth_code uses PKCE. Both desktop and web app scenarios.